x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ssl/RestrictedTrustManagerTests.java

```diff
@@ -40,6 +46,34 @@
     private int numberOfClusters;
     private int numberOfNodes;
 
+    private static Locale restoreLocale;
+
+    @BeforeClass
+    public static void ensureSupportedLocale() throws Exception {
+        Logger logger = Loggers.getLogger(RestrictedTrustManagerTests.class);
+        if (isUnusableLocale()) {
+            // See: https://github.com/elastic/elasticsearch/issues/33081
+            logger.warn("Attempting to run RestrictedTrustManagerTests tests in an unusable locale in a FIPS JVM. Certificate expiration " +
+                "validation will fail, switching to English");
+            restoreLocale = Locale.getDefault();
+            Locale.setDefault(Locale.ENGLISH);
+        }
+    }
+
+    private static boolean isUnusableLocale() {
+        return inFipsJvm() && (Locale.getDefault().toLanguageTag().equals("th-TH")
+            || Locale.getDefault().toLanguageTag().equals("ja-JP-u-ca-japanese-x-lvariant-JP")
+            || Locale.getDefault().toLanguageTag().equals("th-TH-u-nu-thai-x-lvariant-TH"));
+    }
```

Comment:
Those are _very_ specific. I trust you've done the analysis and that's exactly what we need?

Reply:
Yes, see https://github.com/elastic/elasticsearch/issues/33081 and https://github.com/bcgit/bc-java/issues/405